// @wix/auto_sdk_data_permissions/build/es/src/data-v1-data-permissions-permissions.public.js
import { addSpecialPermissions as universalAddSpecialPermissions, getMyPermissions as universalGetMyPermissions, getPermissions as universalGetPermissions, removeSpecialPermissions as universalRemoveSpecialPermissions, updatePermissions as universalUpdatePermissions, updateSpecialPermissions as universalUpdateSpecialPermissions, } from './data-v1-data-permissions-permissions.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/data' };
export function getPermissions(httpClient) {
    return (dataCollectionId) => universalGetPermissions(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
export function updatePermissions(httpClient) {
    return (dataPermissions) => universalUpdatePermissions(dataPermissions, 
    // @ts-ignore
    { httpClient });
}
export function addSpecialPermissions(httpClient) {
    return (dataCollectionId, options) => universalAddSpecialPermissions(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
export function updateSpecialPermissions(httpClient) {
    return (_id, specialPermissions) => universalUpdateSpecialPermissions(_id, specialPermissions, 
    // @ts-ignore
    { httpClient });
}
export function removeSpecialPermissions(httpClient) {
    return (specialPermissionsId) => universalRemoveSpecialPermissions(specialPermissionsId, 
    // @ts-ignore
    { httpClient });
}
export function getMyPermissions(httpClient) {
    return (dataCollectionId) => universalGetMyPermissions(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
export { Access, AccessLevel, Effect, JoinedConditionOperator, PolicyOwnerType, PolicyType, SimpleConditionOperator, SubjectContextType, SubjectType, } from './data-v1-data-permissions-permissions.universal.js';
