// @wix/auto_sdk_data_external-database-connections/build/cjs/src/data-v1-external-database-connection-external-database-connections.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.ProtocolVersion = exports.FieldType = exports.ConnectionType = exports.CollectionsFound = exports.CauseOfFailure = exports.deleteExternalDatabaseConnection = exports.updateExternalDatabaseConnection = exports.createExternalDatabaseConnection = exports.listExternalDatabaseConnections = exports.getExternalDatabaseConnection = exports.__metadata = void 0;
const data_v1_external_database_connection_external_database_connections_universal_js_1 = require("./data-v1-external-database-connection-external-database-connections.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function getExternalDatabaseConnection(httpClient) {
    return (name) => (0, data_v1_external_database_connection_external_database_connections_universal_js_1.getExternalDatabaseConnection)(name, 
    // @ts-ignore
    { httpClient });
}
exports.getExternalDatabaseConnection = getExternalDatabaseConnection;
function listExternalDatabaseConnections(httpClient) {
    return (options) => (0, data_v1_external_database_connection_external_database_connections_universal_js_1.listExternalDatabaseConnections)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listExternalDatabaseConnections = listExternalDatabaseConnections;
function createExternalDatabaseConnection(httpClient) {
    return (externalDatabaseConnection, connectionType) => (0, data_v1_external_database_connection_external_database_connections_universal_js_1.createExternalDatabaseConnection)(externalDatabaseConnection, connectionType, 
    // @ts-ignore
    { httpClient });
}
exports.createExternalDatabaseConnection = createExternalDatabaseConnection;
function updateExternalDatabaseConnection(httpClient) {
    return (name, externalDatabaseConnection) => (0, data_v1_external_database_connection_external_database_connections_universal_js_1.updateExternalDatabaseConnection)(name, externalDatabaseConnection, 
    // @ts-ignore
    { httpClient });
}
exports.updateExternalDatabaseConnection = updateExternalDatabaseConnection;
function deleteExternalDatabaseConnection(httpClient) {
    return (name) => (0, data_v1_external_database_connection_external_database_connections_universal_js_1.deleteExternalDatabaseConnection)(name, 
    // @ts-ignore
    { httpClient });
}
exports.deleteExternalDatabaseConnection = deleteExternalDatabaseConnection;
var data_v1_external_database_connection_external_database_connections_universal_js_2 = require("./data-v1-external-database-connection-external-database-connections.universal.js");
Object.defineProperty(exports, "CauseOfFailure", { enumerable: true, get: function () { return data_v1_external_database_connection_external_database_connections_universal_js_2.CauseOfFailure; } });
Object.defineProperty(exports, "CollectionsFound", { enumerable: true, get: function () { return data_v1_external_database_connection_external_database_connections_universal_js_2.CollectionsFound; } });
Object.defineProperty(exports, "ConnectionType", { enumerable: true, get: function () { return data_v1_external_database_connection_external_database_connections_universal_js_2.ConnectionType; } });
Object.defineProperty(exports, "FieldType", { enumerable: true, get: function () { return data_v1_external_database_connection_external_database_connections_universal_js_2.FieldType; } });
Object.defineProperty(exports, "ProtocolVersion", { enumerable: true, get: function () { return data_v1_external_database_connection_external_database_connections_universal_js_2.ProtocolVersion; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return data_v1_external_database_connection_external_database_connections_universal_js_2.WebhookIdentityType; } });
