// @wix/auto_sdk_data_permissions/build/internal/es/src/data-v1-data-permissions-permissions.meta.js
import * as ambassadorWixDataV1DataPermissions from './data-v1-data-permissions-permissions.http.js';
export function getPermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.getPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/permissions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updatePermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.updatePermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/permissions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function addSpecialPermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.addSpecialPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/permissions/special',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateSpecialPermissions() {
    const payload = {
        specialPermissions: { id: ':specialPermissionsId' },
    };
    const getRequestOptions = ambassadorWixDataV1DataPermissions.updateSpecialPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/permissions/special/{specialPermissions.id}',
        pathParams: { specialPermissionsId: 'specialPermissionsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function removeSpecialPermissions() {
    const payload = { specialPermissionsId: ':specialPermissionsId' };
    const getRequestOptions = ambassadorWixDataV1DataPermissions.removeSpecialPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/permissions/special/{specialPermissionsId}',
        pathParams: { specialPermissionsId: 'specialPermissionsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getMyPermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.getMyPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/permissions/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
