// @wix/auto_sdk_data_permissions/build/internal/es/src/data-v1-data-permissions-permissions.types.js
/**
 * Describes who can perform certain action.
 * Each level includes all levels below it (except UNDEFINED).
 */
export var AccessLevel;
(function (AccessLevel) {
    /** Not set */
    AccessLevel["UNKNOWN"] = "UNKNOWN";
    /** Any subject, including visitors */
    AccessLevel["ANYONE"] = "ANYONE";
    /** Any signed-in user (both site members and collaborators) */
    AccessLevel["SITE_MEMBER"] = "SITE_MEMBER";
    /** Any signed-in user, but site members only have access to own items */
    AccessLevel["SITE_MEMBER_AUTHOR"] = "SITE_MEMBER_AUTHOR";
    /** Site collaborator that has a role with CMS Access permission */
    AccessLevel["CMS_EDITOR"] = "CMS_EDITOR";
    /** CMS administrators and users or roles granted with special access */
    AccessLevel["PRIVILEGED"] = "PRIVILEGED";
})(AccessLevel || (AccessLevel = {}));
export var Access;
(function (Access) {
    /** Action is not specifically allowed, but can be allowed by top-level permissions */
    Access["UNSPECIFIED"] = "UNSPECIFIED";
    /** Action is specifically allowed */
    Access["ALLOWED"] = "ALLOWED";
})(Access || (Access = {}));
export var SubjectType;
(function (SubjectType) {
    SubjectType["UNKNOWN"] = "UNKNOWN";
    SubjectType["ACCOUNT"] = "ACCOUNT";
    SubjectType["USER"] = "USER";
    SubjectType["USER_GROUP"] = "USER_GROUP";
    SubjectType["MEMBER_GROUP"] = "MEMBER_GROUP";
    SubjectType["VISITOR_GROUP"] = "VISITOR_GROUP";
    SubjectType["EXTERNAL_APP"] = "EXTERNAL_APP";
    SubjectType["ACCOUNT_GROUP"] = "ACCOUNT_GROUP";
    SubjectType["WIX_APP"] = "WIX_APP";
})(SubjectType || (SubjectType = {}));
export var SubjectContextType;
(function (SubjectContextType) {
    SubjectContextType["UNKNOWN_CTX"] = "UNKNOWN_CTX";
    SubjectContextType["ORG_CTX"] = "ORG_CTX";
    SubjectContextType["ACCOUNT_CTX"] = "ACCOUNT_CTX";
})(SubjectContextType || (SubjectContextType = {}));
export var SimpleConditionOperator;
(function (SimpleConditionOperator) {
    SimpleConditionOperator["UNKNOWN_SIMPLE_OP"] = "UNKNOWN_SIMPLE_OP";
    SimpleConditionOperator["EQUAL"] = "EQUAL";
})(SimpleConditionOperator || (SimpleConditionOperator = {}));
export var JoinedConditionOperator;
(function (JoinedConditionOperator) {
    JoinedConditionOperator["UNKNOWN_JOIN_OP"] = "UNKNOWN_JOIN_OP";
    JoinedConditionOperator["OR"] = "OR";
    JoinedConditionOperator["AND"] = "AND";
})(JoinedConditionOperator || (JoinedConditionOperator = {}));
export var Effect;
(function (Effect) {
    Effect["UNKNOWN_EFFECT"] = "UNKNOWN_EFFECT";
    Effect["ALLOW"] = "ALLOW";
    Effect["DENY"] = "DENY";
})(Effect || (Effect = {}));
export var PolicyType;
(function (PolicyType) {
    PolicyType["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
    PolicyType["PREDEFINED"] = "PREDEFINED";
    PolicyType["CUSTOM"] = "CUSTOM";
    PolicyType["INLINE_CUSTOM"] = "INLINE_CUSTOM";
})(PolicyType || (PolicyType = {}));
export var PolicyOwnerType;
(function (PolicyOwnerType) {
    PolicyOwnerType["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
    PolicyOwnerType["WIX"] = "WIX";
    PolicyOwnerType["WIX_ACCOUNT"] = "WIX_ACCOUNT";
})(PolicyOwnerType || (PolicyOwnerType = {}));
