// @wix/auto_sdk_data_permissions/build/internal/cjs/src/data-v1-data-permissions-permissions.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SubjectType = exports.SubjectContextType = exports.SimpleConditionOperator = exports.PolicyType = exports.PolicyOwnerType = exports.JoinedConditionOperator = exports.Effect = exports.AccessLevel = exports.Access = exports.getMyPermissions = exports.removeSpecialPermissions = exports.updateSpecialPermissions = exports.addSpecialPermissions = exports.updatePermissions = exports.getPermissions = exports.__metadata = void 0;
const data_v1_data_permissions_permissions_universal_js_1 = require("./data-v1-data-permissions-permissions.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function getPermissions(httpClient) {
    return (dataCollectionId) => (0, data_v1_data_permissions_permissions_universal_js_1.getPermissions)(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
exports.getPermissions = getPermissions;
function updatePermissions(httpClient) {
    return (dataPermissions) => (0, data_v1_data_permissions_permissions_universal_js_1.updatePermissions)(dataPermissions, 
    // @ts-ignore
    { httpClient });
}
exports.updatePermissions = updatePermissions;
function addSpecialPermissions(httpClient) {
    return (dataCollectionId, options) => (0, data_v1_data_permissions_permissions_universal_js_1.addSpecialPermissions)(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.addSpecialPermissions = addSpecialPermissions;
function updateSpecialPermissions(httpClient) {
    return (_id, specialPermissions) => (0, data_v1_data_permissions_permissions_universal_js_1.updateSpecialPermissions)(_id, specialPermissions, 
    // @ts-ignore
    { httpClient });
}
exports.updateSpecialPermissions = updateSpecialPermissions;
function removeSpecialPermissions(httpClient) {
    return (specialPermissionsId) => (0, data_v1_data_permissions_permissions_universal_js_1.removeSpecialPermissions)(specialPermissionsId, 
    // @ts-ignore
    { httpClient });
}
exports.removeSpecialPermissions = removeSpecialPermissions;
function getMyPermissions(httpClient) {
    return (dataCollectionId) => (0, data_v1_data_permissions_permissions_universal_js_1.getMyPermissions)(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
exports.getMyPermissions = getMyPermissions;
var data_v1_data_permissions_permissions_universal_js_2 = require("./data-v1-data-permissions-permissions.universal.js");
Object.defineProperty(exports, "Access", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.Access; } });
Object.defineProperty(exports, "AccessLevel", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.AccessLevel; } });
Object.defineProperty(exports, "Effect", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.Effect; } });
Object.defineProperty(exports, "JoinedConditionOperator", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.JoinedConditionOperator; } });
Object.defineProperty(exports, "PolicyOwnerType", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.PolicyOwnerType; } });
Object.defineProperty(exports, "PolicyType", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.PolicyType; } });
Object.defineProperty(exports, "SimpleConditionOperator", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.SimpleConditionOperator; } });
Object.defineProperty(exports, "SubjectContextType", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.SubjectContextType; } });
Object.defineProperty(exports, "SubjectType", { enumerable: true, get: function () { return data_v1_data_permissions_permissions_universal_js_2.SubjectType; } });
