// @wix/auto_sdk_data_permissions/build/internal/cjs/src/data-v1-data-permissions-permissions.universal.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMyPermissions = exports.removeSpecialPermissions = exports.updateSpecialPermissions = exports.addSpecialPermissions = exports.updatePermissions = exports.getPermissions = exports.PolicyOwnerType = exports.PolicyType = exports.Effect = exports.JoinedConditionOperator = exports.SimpleConditionOperator = exports.SubjectContextType = exports.SubjectType = exports.Access = exports.AccessLevel = void 0;
const transform_error_1 = require("@wix/sdk-runtime/transform-error");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const ambassadorWixDataV1DataPermissions = __importStar(require("./data-v1-data-permissions-permissions.http.js"));
/**
 * Describes who can perform certain action.
 * Each level includes all levels below it (except UNDEFINED).
 */
var AccessLevel;
(function (AccessLevel) {
    /** Not set */
    AccessLevel["UNKNOWN"] = "UNKNOWN";
    /** Any subject, including visitors */
    AccessLevel["ANYONE"] = "ANYONE";
    /** Any signed-in user (both site members and collaborators) */
    AccessLevel["SITE_MEMBER"] = "SITE_MEMBER";
    /** Any signed-in user, but site members only have access to own items */
    AccessLevel["SITE_MEMBER_AUTHOR"] = "SITE_MEMBER_AUTHOR";
    /** Site collaborator that has a role with CMS Access permission */
    AccessLevel["CMS_EDITOR"] = "CMS_EDITOR";
    /** CMS administrators and users or roles granted with special access */
    AccessLevel["PRIVILEGED"] = "PRIVILEGED";
})(AccessLevel || (exports.AccessLevel = AccessLevel = {}));
var Access;
(function (Access) {
    /** Action is not specifically allowed, but can be allowed by top-level permissions */
    Access["UNSPECIFIED"] = "UNSPECIFIED";
    /** Action is specifically allowed */
    Access["ALLOWED"] = "ALLOWED";
})(Access || (exports.Access = Access = {}));
var SubjectType;
(function (SubjectType) {
    SubjectType["UNKNOWN"] = "UNKNOWN";
    SubjectType["ACCOUNT"] = "ACCOUNT";
    SubjectType["USER"] = "USER";
    SubjectType["USER_GROUP"] = "USER_GROUP";
    SubjectType["MEMBER_GROUP"] = "MEMBER_GROUP";
    SubjectType["VISITOR_GROUP"] = "VISITOR_GROUP";
    SubjectType["EXTERNAL_APP"] = "EXTERNAL_APP";
    SubjectType["ACCOUNT_GROUP"] = "ACCOUNT_GROUP";
    SubjectType["WIX_APP"] = "WIX_APP";
})(SubjectType || (exports.SubjectType = SubjectType = {}));
var SubjectContextType;
(function (SubjectContextType) {
    SubjectContextType["UNKNOWN_CTX"] = "UNKNOWN_CTX";
    SubjectContextType["ORG_CTX"] = "ORG_CTX";
    SubjectContextType["ACCOUNT_CTX"] = "ACCOUNT_CTX";
})(SubjectContextType || (exports.SubjectContextType = SubjectContextType = {}));
var SimpleConditionOperator;
(function (SimpleConditionOperator) {
    SimpleConditionOperator["UNKNOWN_SIMPLE_OP"] = "UNKNOWN_SIMPLE_OP";
    SimpleConditionOperator["EQUAL"] = "EQUAL";
})(SimpleConditionOperator || (exports.SimpleConditionOperator = SimpleConditionOperator = {}));
var JoinedConditionOperator;
(function (JoinedConditionOperator) {
    JoinedConditionOperator["UNKNOWN_JOIN_OP"] = "UNKNOWN_JOIN_OP";
    JoinedConditionOperator["OR"] = "OR";
    JoinedConditionOperator["AND"] = "AND";
})(JoinedConditionOperator || (exports.JoinedConditionOperator = JoinedConditionOperator = {}));
var Effect;
(function (Effect) {
    Effect["UNKNOWN_EFFECT"] = "UNKNOWN_EFFECT";
    Effect["ALLOW"] = "ALLOW";
    Effect["DENY"] = "DENY";
})(Effect || (exports.Effect = Effect = {}));
var PolicyType;
(function (PolicyType) {
    PolicyType["UNKNOWN_STATUS"] = "UNKNOWN_STATUS";
    PolicyType["PREDEFINED"] = "PREDEFINED";
    PolicyType["CUSTOM"] = "CUSTOM";
    PolicyType["INLINE_CUSTOM"] = "INLINE_CUSTOM";
})(PolicyType || (exports.PolicyType = PolicyType = {}));
var PolicyOwnerType;
(function (PolicyOwnerType) {
    PolicyOwnerType["UNKNOWN_TYPE"] = "UNKNOWN_TYPE";
    PolicyOwnerType["WIX"] = "WIX";
    PolicyOwnerType["WIX_ACCOUNT"] = "WIX_ACCOUNT";
})(PolicyOwnerType || (exports.PolicyOwnerType = PolicyOwnerType = {}));
/**
 * Returns data permissions for a data collection
 * @param dataCollectionId - Data Collection ID to get permissions for
 * @public
 * @documentationMaturity preview
 * @requiredField dataCollectionId
 * @permissionId WIX_DATA.PERMISSIONS_READ
 * @permissionScope Manage All Data Resources
 * @permissionScopeId SCOPE.DC-DATA.MANAGE-ALL
 * @permissionScope Manage Data Collections
 * @permissionScopeId SCOPE.DC-DATA.DATA-COLLECTIONS-MANAGE
 * @applicableIdentity APP
 * @fqn wix.cloud.permissions.v1.DataPermissionsService.GetPermissions
 */
async function getPermissions(dataCollectionId) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        dataCollectionId: dataCollectionId,
    });
    const reqOpts = ambassadorWixDataV1DataPermissions.getPermissions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { dataCollectionId: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['dataCollectionId']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.getPermissions = getPermissions;
/**
 * Updates data permissions for a data collection
 *
 * NOTE that special permissions are not updated using this API, use dedicated methods for it.
 * @param dataPermissions - Data permissions to update
 * @public
 * @documentationMaturity preview
 * @requiredField dataPermissions
 * @requiredField dataPermissions._id
 * @requiredField dataPermissions.itemInsert
 * @requiredField dataPermissions.itemRead
 * @requiredField dataPermissions.itemRemove
 * @requiredField dataPermissions.itemUpdate
 * @permissionId WIX_DATA.PERMISSIONS_UPDATE
 * @permissionScope Manage All Data Resources
 * @permissionScopeId SCOPE.DC-DATA.MANAGE-ALL
 * @permissionScope Manage Data Collections
 * @permissionScopeId SCOPE.DC-DATA.DATA-COLLECTIONS-MANAGE
 * @applicableIdentity APP
 * @fqn wix.cloud.permissions.v1.DataPermissionsService.UpdatePermissions
 */
async function updatePermissions(dataPermissions) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        dataPermissions: dataPermissions,
    });
    const reqOpts = ambassadorWixDataV1DataPermissions.updatePermissions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { dataPermissions: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['dataPermissions']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.updatePermissions = updatePermissions;
/**
 * Adds special data permissions for particular user/group
 * Errors:
 * - `ALREADY_EXISTS` in case if special permission for same subject already exists
 * @param dataCollectionId - Data Collection ID to add special permission for
 * @public
 * @documentationMaturity preview
 * @requiredField dataCollectionId
 * @requiredField options
 * @requiredField options.specialPermissions
 * @requiredField options.specialPermissions.subject
 * @permissionId WIX_DATA.PERMISSIONS_UPDATE_SPECIAL
 * @permissionScope Manage All Data Resources
 * @permissionScopeId SCOPE.DC-DATA.MANAGE-ALL
 * @permissionScope Manage Data Collections
 * @permissionScopeId SCOPE.DC-DATA.DATA-COLLECTIONS-MANAGE
 * @applicableIdentity APP
 * @fqn wix.cloud.permissions.v1.DataPermissionsService.AddSpecialPermissions
 */
async function addSpecialPermissions(dataCollectionId, options) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        dataCollectionId: dataCollectionId,
        specialPermissions: options?.specialPermissions,
    });
    const reqOpts = ambassadorWixDataV1DataPermissions.addSpecialPermissions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: {
                dataCollectionId: '$[0]',
                specialPermissions: '$[1].specialPermissions',
            },
            singleArgumentUnchanged: false,
        }, ['dataCollectionId', 'options']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.addSpecialPermissions = addSpecialPermissions;
/**
 * Updates special data permissions for particular user/group.
 * Any omitted permissions will be set to UNSPECIFIED.
 * @param _id - ID of this special access, so it can be managed separately
 * @public
 * @documentationMaturity preview
 * @requiredField _id
 * @requiredField specialPermissions
 * @permissionId WIX_DATA.PERMISSIONS_UPDATE_SPECIAL
 * @permissionScope Manage All Data Resources
 * @permissionScopeId SCOPE.DC-DATA.MANAGE-ALL
 * @permissionScope Manage Data Collections
 * @permissionScopeId SCOPE.DC-DATA.DATA-COLLECTIONS-MANAGE
 * @applicableIdentity APP
 * @fqn wix.cloud.permissions.v1.DataPermissionsService.UpdateSpecialPermissions
 */
async function updateSpecialPermissions(_id, specialPermissions) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[2];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        specialPermissions: { ...specialPermissions, id: _id },
    });
    const reqOpts = ambassadorWixDataV1DataPermissions.updateSpecialPermissions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: { specialPermissions: '$[1]' },
            explicitPathsToArguments: { 'specialPermissions.id': '$[0]' },
            singleArgumentUnchanged: false,
        }, ['_id', 'specialPermissions']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.updateSpecialPermissions = updateSpecialPermissions;
/**
 * Deletes special data permissions for particular user/group
 * @param specialPermissionsId - Special permissions ID to remove
 * @public
 * @documentationMaturity preview
 * @requiredField specialPermissionsId
 * @permissionId WIX_DATA.PERMISSIONS_UPDATE_SPECIAL
 * @permissionScope Manage All Data Resources
 * @permissionScopeId SCOPE.DC-DATA.MANAGE-ALL
 * @permissionScope Manage Data Collections
 * @permissionScopeId SCOPE.DC-DATA.DATA-COLLECTIONS-MANAGE
 * @applicableIdentity APP
 * @fqn wix.cloud.permissions.v1.DataPermissionsService.RemoveSpecialPermissions
 */
async function removeSpecialPermissions(specialPermissionsId) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        specialPermissionsId: specialPermissionsId,
    });
    const reqOpts = ambassadorWixDataV1DataPermissions.removeSpecialPermissions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { specialPermissionsId: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['specialPermissionsId']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.removeSpecialPermissions = removeSpecialPermissions;
/**
 * Returns current user permissions for given data collection
 * @param dataCollectionId - Data Collection ID to get permission for
 * @public
 * @documentationMaturity preview
 * @requiredField dataCollectionId
 * @permissionId WIX_DATA.PERMISSIONS_GET_ALLOWED
 * @permissionScope Manage All Data Resources
 * @permissionScopeId SCOPE.DC-DATA.MANAGE-ALL
 * @permissionScope Manage Data Collections
 * @permissionScopeId SCOPE.DC-DATA.DATA-COLLECTIONS-MANAGE
 * @applicableIdentity APP
 * @fqn wix.cloud.permissions.v1.DataPermissionsService.GetMyPermissions
 */
async function getMyPermissions(dataCollectionId) {
    // @ts-ignore
    const { httpClient, sideEffects } = arguments[1];
    const payload = (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)({
        dataCollectionId: dataCollectionId,
    });
    const reqOpts = ambassadorWixDataV1DataPermissions.getMyPermissions(payload);
    sideEffects?.onSiteCall?.();
    try {
        const result = await httpClient.request(reqOpts);
        sideEffects?.onSuccess?.(result);
        return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(result.data);
    }
    catch (err) {
        const transformedError = (0, transform_error_1.transformError)(err, {
            spreadPathsToArguments: {},
            explicitPathsToArguments: { dataCollectionId: '$[0]' },
            singleArgumentUnchanged: false,
        }, ['dataCollectionId']);
        sideEffects?.onError?.(err);
        throw transformedError;
    }
}
exports.getMyPermissions = getMyPermissions;
