// @wix/auto_sdk_data_external-database-connections/build/es/src/data-v1-external-database-connection-external-database-connections.public.js
import { createExternalDatabaseConnection as universalCreateExternalDatabaseConnection, deleteExternalDatabaseConnection as universalDeleteExternalDatabaseConnection, getExternalDatabaseConnection as universalGetExternalDatabaseConnection, listExternalDatabaseConnections as universalListExternalDatabaseConnections, updateExternalDatabaseConnection as universalUpdateExternalDatabaseConnection, } from './data-v1-external-database-connection-external-database-connections.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/data' };
export function getExternalDatabaseConnection(httpClient) {
    return (name) => universalGetExternalDatabaseConnection(name, 
    // @ts-ignore
    { httpClient });
}
export function listExternalDatabaseConnections(httpClient) {
    return (options) => universalListExternalDatabaseConnections(options, 
    // @ts-ignore
    { httpClient });
}
export function createExternalDatabaseConnection(httpClient) {
    return (externalDatabaseConnection, connectionType) => universalCreateExternalDatabaseConnection(externalDatabaseConnection, connectionType, 
    // @ts-ignore
    { httpClient });
}
export function updateExternalDatabaseConnection(httpClient) {
    return (name, externalDatabaseConnection) => universalUpdateExternalDatabaseConnection(name, externalDatabaseConnection, 
    // @ts-ignore
    { httpClient });
}
export function deleteExternalDatabaseConnection(httpClient) {
    return (name) => universalDeleteExternalDatabaseConnection(name, 
    // @ts-ignore
    { httpClient });
}
export { CauseOfFailure, CollectionsFound, ConnectionType, FieldType, ProtocolVersion, WebhookIdentityType, } from './data-v1-external-database-connection-external-database-connections.universal.js';
