// @wix/auto_sdk_data_indexes/build/internal/es/src/data-v2-index-indexes.meta.js
import * as ambassadorWixDataV2Index from './data-v2-index-indexes.http.js';
export function createIndex() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Index.createIndex(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/indexes',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function dropIndex() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Index.dropIndex(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/indexes',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listIndexes() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Index.listIndexes(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/indexes',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listAvailableIndexes() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Index.listAvailableIndexes(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/indexes/available',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
