// @wix/auto_sdk_data_indexes/build/internal/es/src/data-v2-index-indexes.types.js
/**
 * Order determines how values are ordered in the index. This is important when
 * ordering and/or range querying by indexed fields.
 */
export var Order;
(function (Order) {
    Order["ASC"] = "ASC";
    Order["DESC"] = "DESC";
})(Order || (Order = {}));
export var Status;
(function (Status) {
    /** Place holder. Never returned by the service. */
    Status["UNKNOWN"] = "UNKNOWN";
    /** Index creation is in progress. */
    Status["BUILDING"] = "BUILDING";
    /** Index has been successfully created and can be used in queries. */
    Status["ACTIVE"] = "ACTIVE";
    /** Index is in the process of being dropped. */
    Status["DROPPING"] = "DROPPING";
    /** Index has been dropped successfully. */
    Status["DROPPED"] = "DROPPED";
    /** Index creation has failed. */
    Status["FAILED"] = "FAILED";
    /** Index contains incorrectly indexed data. */
    Status["INVALID"] = "INVALID";
})(Status || (Status = {}));
export var IndexSource;
(function (IndexSource) {
    /** Place holder. Never returned by the service. */
    IndexSource["UNKNOWN_INDEX_SOURCE"] = "UNKNOWN_INDEX_SOURCE";
    /** Index is created by the system, present for all collections. */
    IndexSource["SYSTEM"] = "SYSTEM";
    /** Index was created by a user via API or UI. */
    IndexSource["USER"] = "USER";
    /** Index is created automatically by the system based on user queries. */
    IndexSource["AUTO"] = "AUTO";
})(IndexSource || (IndexSource = {}));
export var Environment;
(function (Environment) {
    /** Place holder. Never returned by the service. */
    Environment["UNKNOWN_ENVIRONMENT"] = "UNKNOWN_ENVIRONMENT";
    /** Collection [sandbox environment](https://support.wix.com/en/article/cms-about-sandbox-and-live-collections-and-syncing#sandbox-collections). */
    Environment["SANDBOX"] = "SANDBOX";
    /** If the [collection's sandbox environment is enabled](https://support.wix.com/en/article/cms-accessing-and-syncing-your-sandbox-and-live-collections#enabling-sandbox-collections), run the task in the collection's sandbox environment. If not, run the task in the collection's live environment. */
    Environment["SANDBOX_PREFERRED"] = "SANDBOX_PREFERRED";
    /** Collection [live environment](https://support.wix.com/en/article/cms-about-sandbox-and-live-collections-and-syncing#live-collections). */
    Environment["LIVE"] = "LIVE";
})(Environment || (Environment = {}));
