// @wix/auto_sdk_data_external-database/build/cjs/src/service-plugins-error-classes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BadRequestWixError = exports.CollectionChangeNotSupportedWixError = exports.ValidationWixError = exports.ReferenceNotFoundWixError = exports.ReferenceAlreadyExistsWixError = exports.CollectionNotFoundWixError = exports.CollectionAlreadyExistsWixError = exports.ItemNotFoundWixError = exports.ItemAlreadyExistsWixError = void 0;
class ItemAlreadyExistsWixError extends Error {
    constructor(data = {}) {
        super('ItemAlreadyExists');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'ITEM_ALREADY_EXISTS';
        this.name = 'ItemAlreadyExists';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.ItemAlreadyExistsError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'ItemAlreadyExists',
            applicationCode: 'ITEM_ALREADY_EXISTS',
        };
        this.data = data;
    }
}
exports.ItemAlreadyExistsWixError = ItemAlreadyExistsWixError;
/** @hidden */
ItemAlreadyExistsWixError.__type = 'wix_spi_error';
class ItemNotFoundWixError extends Error {
    constructor(data = {}) {
        super('ItemNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'ITEM_NOT_FOUND';
        this.name = 'ItemNotFound';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.ItemNotFoundError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'ItemNotFound',
            applicationCode: 'ITEM_NOT_FOUND',
        };
        this.data = data;
    }
}
exports.ItemNotFoundWixError = ItemNotFoundWixError;
/** @hidden */
ItemNotFoundWixError.__type = 'wix_spi_error';
class CollectionAlreadyExistsWixError extends Error {
    constructor(data = {}) {
        super('CollectionAlreadyExists');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'COLLECTION_ALREADY_EXISTS';
        this.name = 'CollectionAlreadyExists';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.CollectionAlreadyExistsError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CollectionAlreadyExists',
            applicationCode: 'COLLECTION_ALREADY_EXISTS',
        };
        this.data = data;
    }
}
exports.CollectionAlreadyExistsWixError = CollectionAlreadyExistsWixError;
/** @hidden */
CollectionAlreadyExistsWixError.__type = 'wix_spi_error';
class CollectionNotFoundWixError extends Error {
    constructor(data = {}) {
        super('CollectionNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'COLLECTION_NOT_FOUND';
        this.name = 'CollectionNotFound';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.CollectionNotFoundError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CollectionNotFound',
            applicationCode: 'COLLECTION_NOT_FOUND',
        };
        this.data = data;
    }
}
exports.CollectionNotFoundWixError = CollectionNotFoundWixError;
/** @hidden */
CollectionNotFoundWixError.__type = 'wix_spi_error';
class ReferenceAlreadyExistsWixError extends Error {
    constructor(data = {}) {
        super('ReferenceAlreadyExists');
        this.httpCode = 409;
        this.statusCode = 'ALREADY_EXISTS';
        this.applicationCode = 'REFERENCE_ALREADY_EXISTS';
        this.name = 'ReferenceAlreadyExists';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.ReferenceAlreadyExistsError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'ReferenceAlreadyExists',
            applicationCode: 'REFERENCE_ALREADY_EXISTS',
        };
        this.data = data;
    }
}
exports.ReferenceAlreadyExistsWixError = ReferenceAlreadyExistsWixError;
/** @hidden */
ReferenceAlreadyExistsWixError.__type = 'wix_spi_error';
class ReferenceNotFoundWixError extends Error {
    constructor(data = {}) {
        super('ReferenceNotFound');
        this.httpCode = 404;
        this.statusCode = 'NOT_FOUND';
        this.applicationCode = 'REFERENCE_NOT_FOUND';
        this.name = 'ReferenceNotFound';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.ReferenceNotFoundError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'ReferenceNotFound',
            applicationCode: 'REFERENCE_NOT_FOUND',
        };
        this.data = data;
    }
}
exports.ReferenceNotFoundWixError = ReferenceNotFoundWixError;
/** @hidden */
ReferenceNotFoundWixError.__type = 'wix_spi_error';
class ValidationWixError extends Error {
    constructor(data = {}) {
        super('ValidationError');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'VALIDATION_ERROR';
        this.name = 'ValidationError';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.ValidationError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'ValidationError',
            applicationCode: 'VALIDATION_ERROR',
        };
        this.data = data;
    }
}
exports.ValidationWixError = ValidationWixError;
/** @hidden */
ValidationWixError.__type = 'wix_spi_error';
class CollectionChangeNotSupportedWixError extends Error {
    constructor(data = {}) {
        super('CollectionChangeNotSupportedError');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'COLLECTION_CHANGE_NOT_SUPPORTED';
        this.name = 'CollectionChangeNotSupportedError';
        this.errorSchemaName =
            'com.wixpress.cloud.externaldatabasespi.v3.CollectionChangeNotSupported';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'CollectionChangeNotSupportedError',
            applicationCode: 'COLLECTION_CHANGE_NOT_SUPPORTED',
        };
        this.data = data;
    }
}
exports.CollectionChangeNotSupportedWixError = CollectionChangeNotSupportedWixError;
/** @hidden */
CollectionChangeNotSupportedWixError.__type = 'wix_spi_error';
class BadRequestWixError extends Error {
    constructor() {
        super('BadRequestError');
        this.httpCode = 400;
        this.statusCode = 'INVALID_ARGUMENT';
        this.applicationCode = 'BAD_REQUEST';
        this.name = 'BadRequestError';
        this.errorType = 'SPI';
        this.spiErrorData = {
            name: 'BadRequestError',
            applicationCode: 'BAD_REQUEST',
        };
    }
}
exports.BadRequestWixError = BadRequestWixError;
/** @hidden */
BadRequestWixError.__type = 'wix_spi_error';
