// @wix/auto_sdk_data_movement-job/build/internal/cjs/src/hub-v1-job-movement-job.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.listMovementLogs = exports.terminateJob = exports.listJobs = exports.getJob = exports.createFileUploadUrl = exports.submitJob = void 0;
const ambassadorWixHubV1Job = __importStar(require("./hub-v1-job-movement-job.http.js"));
function submitJob() {
    const payload = {};
    const getRequestOptions = ambassadorWixHubV1Job.submitJob(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/jobs',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.submitJob = submitJob;
function createFileUploadUrl() {
    const payload = {};
    const getRequestOptions = ambassadorWixHubV1Job.createFileUploadUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/temporary-files/upload-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createFileUploadUrl = createFileUploadUrl;
function getJob() {
    const payload = { jobId: ':jobId' };
    const getRequestOptions = ambassadorWixHubV1Job.getJob(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/jobs/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getJob = getJob;
function listJobs() {
    const payload = {};
    const getRequestOptions = ambassadorWixHubV1Job.listJobs(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/jobs',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listJobs = listJobs;
function terminateJob() {
    const payload = { jobId: ':jobId' };
    const getRequestOptions = ambassadorWixHubV1Job.terminateJob(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/jobs/{jobId}/terminate',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.terminateJob = terminateJob;
function listMovementLogs() {
    const payload = { jobId: ':jobId' };
    const getRequestOptions = ambassadorWixHubV1Job.listMovementLogs(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/jobs/{jobId}/logs',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listMovementLogs = listMovementLogs;
