// @wix/auto_sdk_data_external-database/build/internal/cjs/src/data-externaldatabasespi-v3-main-entity.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.provideHandlers = void 0;
const sdk_types_1 = require("@wix/sdk-types");
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
exports.provideHandlers = (0, sdk_types_1.ServicePluginDefinition)('EXTERNAL_DATABASE_PROVIDER', [
    {
        name: 'queryDataItems',
        primaryHttpMappingPath: '/v3/items/query',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'countDataItems',
        primaryHttpMappingPath: '/v3/items/count',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'aggregateDataItems',
        primaryHttpMappingPath: '/v3/items/aggregate',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'queryDistinctValues',
        primaryHttpMappingPath: '/v3/items/query-distinct-values',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'insertDataItems',
        primaryHttpMappingPath: '/v3/items/insert',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'updateDataItems',
        primaryHttpMappingPath: '/v3/items/update',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'removeDataItems',
        primaryHttpMappingPath: '/v3/items/remove',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'truncateDataItems',
        primaryHttpMappingPath: '/v3/items/truncate',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'queryReferencedDataItems',
        primaryHttpMappingPath: '/v3/items/query-referenced',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'insertDataItemReferences',
        primaryHttpMappingPath: '/v3/items/insert-references',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'removeDataItemReferences',
        primaryHttpMappingPath: '/v3/items/remove-references',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'listCollections',
        primaryHttpMappingPath: '/v3/collections/get',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'createCollection',
        primaryHttpMappingPath: '/v3/collections/create',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'updateCollection',
        primaryHttpMappingPath: '/v3/collections/update',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'deleteCollection',
        primaryHttpMappingPath: '/v3/collections/delete',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
    {
        name: 'getCapabilities',
        primaryHttpMappingPath: '/v3/capabilities/get',
        transformations: {
            toREST: (payload) => {
                const toRestResponse = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromSDKRequestToRESTRequest)(toRestResponse);
            },
            fromREST: (payload) => {
                const fromRestRequest = payload;
                return (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)(fromRestRequest);
            },
        },
    },
]);
