// @wix/auto_sdk_data_backups/build/internal/es/src/data-v2-backup-backups.public.js
import { createBackup as universalCreateBackup, deleteBackup as universalDeleteBackup, listBackups as universalListBackups, listRestorations as universalListRestorations, restoreBackup as universalRestoreBackup, restorePartialBackup as universalRestorePartialBackup, } from './data-v2-backup-backups.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/data' };
export function createBackup(httpClient) {
    return () => universalCreateBackup(
    // @ts-ignore
    { httpClient });
}
export function listBackups(httpClient) {
    return (options) => universalListBackups(options, 
    // @ts-ignore
    { httpClient });
}
export function restoreBackup(httpClient) {
    return (backupId) => universalRestoreBackup(backupId, 
    // @ts-ignore
    { httpClient });
}
export function restorePartialBackup(httpClient) {
    return (backupId, restorationCollections) => universalRestorePartialBackup(backupId, restorationCollections, 
    // @ts-ignore
    { httpClient });
}
export function listRestorations(httpClient) {
    return (options) => universalListRestorations(options, 
    // @ts-ignore
    { httpClient });
}
export function deleteBackup(httpClient) {
    return (backupId) => universalDeleteBackup(backupId, 
    // @ts-ignore
    { httpClient });
}
export { RestorationStatus, Status, Type, WebhookIdentityType, } from './data-v2-backup-backups.universal.js';
