// @wix/auto_sdk_data_backups/build/internal/es/src/data-v2-backup-backups.meta.js
import * as ambassadorWixDataV2Backup from './data-v2-backup-backups.http.js';
export function createBackup() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Backup.createBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/backups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listBackups() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Backup.listBackups(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/backups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function restoreBackup() {
    const payload = { backupId: ':backupId' };
    const getRequestOptions = ambassadorWixDataV2Backup.restoreBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/backups/{backupId}/restore',
        pathParams: { backupId: 'backupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function restorePartialBackup() {
    const payload = { backupId: ':backupId' };
    const getRequestOptions = ambassadorWixDataV2Backup.restorePartialBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/backups/{backupId}/partial-restore',
        pathParams: { backupId: 'backupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listRestorations() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Backup.listRestorations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/backups/restorations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteBackup() {
    const payload = { backupId: ':backupId' };
    const getRequestOptions = ambassadorWixDataV2Backup.deleteBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/backups/{backupId}',
        pathParams: { backupId: 'backupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
