// @wix/auto_sdk_data_collections/build/internal/es/src/data-v2-data-collection-collections.meta.js
import * as ambassadorWixDataV2DataCollection from './data-v2-data-collection-collections.http.js';
export function createDataCollection() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.createDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getDataCollection() {
    const payload = { dataCollectionId: ':dataCollectionId' };
    const getRequestOptions = ambassadorWixDataV2DataCollection.getDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections/{dataCollectionId}',
        pathParams: { dataCollectionId: 'dataCollectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listDataCollections() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.listDataCollections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDataCollection() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.updateDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v2/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteDataCollection() {
    const payload = { dataCollectionId: ':dataCollectionId' };
    const getRequestOptions = ambassadorWixDataV2DataCollection.deleteDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/collections/{dataCollectionId}',
        pathParams: { dataCollectionId: 'dataCollectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function restoreDataCollection() {
    const payload = { dataCollectionId: ':dataCollectionId' };
    const getRequestOptions = ambassadorWixDataV2DataCollection.restoreDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/restore/{dataCollectionId}',
        pathParams: { dataCollectionId: 'dataCollectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createDataCollectionField() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.createDataCollectionField(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/create-field',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function updateDataCollectionField() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.updateDataCollectionField(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/update-field',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function deleteDataCollectionField() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.deleteDataCollectionField(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/delete-field',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
