// @wix/auto_sdk_data_indexes/build/internal/cjs/src/data-v2-index-indexes.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Status = exports.Order = exports.IndexSource = exports.Environment = exports.listAvailableIndexes = exports.listIndexes = exports.dropIndex = exports.createIndex = exports.__metadata = void 0;
const data_v2_index_indexes_universal_js_1 = require("./data-v2-index-indexes.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function createIndex(httpClient) {
    return (dataCollectionId, index) => (0, data_v2_index_indexes_universal_js_1.createIndex)(dataCollectionId, index, 
    // @ts-ignore
    { httpClient });
}
exports.createIndex = createIndex;
function dropIndex(httpClient) {
    return (dataCollectionId, indexName) => (0, data_v2_index_indexes_universal_js_1.dropIndex)(dataCollectionId, indexName, 
    // @ts-ignore
    { httpClient });
}
exports.dropIndex = dropIndex;
function listIndexes(httpClient) {
    return (dataCollectionId, options) => (0, data_v2_index_indexes_universal_js_1.listIndexes)(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.listIndexes = listIndexes;
/** @internal */
function listAvailableIndexes(httpClient) {
    return (dataCollectionId) => (0, data_v2_index_indexes_universal_js_1.listAvailableIndexes)(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
exports.listAvailableIndexes = listAvailableIndexes;
var data_v2_index_indexes_universal_js_2 = require("./data-v2-index-indexes.universal.js");
Object.defineProperty(exports, "Environment", { enumerable: true, get: function () { return data_v2_index_indexes_universal_js_2.Environment; } });
Object.defineProperty(exports, "IndexSource", { enumerable: true, get: function () { return data_v2_index_indexes_universal_js_2.IndexSource; } });
Object.defineProperty(exports, "Order", { enumerable: true, get: function () { return data_v2_index_indexes_universal_js_2.Order; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return data_v2_index_indexes_universal_js_2.Status; } });
