// @wix/auto_sdk_data_tasks/build/cjs/src/data-autocms-v2-background-task-tasks.http.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cancelTask = exports.getTask = exports.listTasks = exports.createTask = void 0;
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const timestamp_2 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const rest_modules_2 = require("@wix/sdk-runtime/rest-modules");
function resolveWixCloudAutocmsTasksV1TaskServiceUrl(opts) {
    const domainToMappings = {
        'editor._base_domain_': [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        'manage._base_domain_': [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        'www._base_domain_': [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        'api._api_base_domain_': [
            {
                srcPath: '/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/cms/v1/tasks',
                destPath: '/v1',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
        _: [
            {
                srcPath: '/_api/autocms/v1/tasks',
                destPath: '/v1',
            },
        ],
    };
    return (0, rest_modules_2.resolveUrl)(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_data_tasks';
/** Creates and runs a new background task. */
function createTask(payload) {
    function __createTask({ host }) {
        const serializedData = (0, transform_paths_1.transformPaths)(payload, [
            {
                transformFn: timestamp_1.transformSDKTimestampToRESTTimestamp,
                paths: [
                    { path: 'task.startedAt' },
                    { path: 'task.finishedAt' },
                    {
                        path: 'task.updatePublishStatusOptions.scheduleDraftStatusOptions.date',
                    },
                    {
                        path: 'task.updatePublishStatusOptions.schedulePublishedStatusOptions.date',
                    },
                ],
            },
        ]);
        const metadata = {
            entityFqdn: 'wix.data.autocms.v2.background_task',
            method: 'POST',
            methodFqn: 'wix.cloud.autocms.tasks.v1.TaskService.CreateTask',
            packageName: PACKAGE_NAME,
            url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
                protoPath: '/v1',
                data: serializedData,
                host,
            }),
            data: serializedData,
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'task.startedAt' },
                        { path: 'task.finishedAt' },
                        {
                            path: 'task.updatePublishStatusOptions.scheduleDraftStatusOptions.date',
                        },
                        {
                            path: 'task.updatePublishStatusOptions.schedulePublishedStatusOptions.date',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __createTask;
}
exports.createTask = createTask;
/** Lists all existing tasks. */
function listTasks(payload) {
    function __listTasks({ host }) {
        const metadata = {
            entityFqdn: 'wix.data.autocms.v2.background_task',
            method: 'GET',
            methodFqn: 'wix.cloud.autocms.tasks.v1.TaskService.ListTasks',
            packageName: PACKAGE_NAME,
            url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
                protoPath: '/v1',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'tasks.startedAt' },
                        { path: 'tasks.finishedAt' },
                        {
                            path: 'tasks.updatePublishStatusOptions.scheduleDraftStatusOptions.date',
                        },
                        {
                            path: 'tasks.updatePublishStatusOptions.schedulePublishedStatusOptions.date',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __listTasks;
}
exports.listTasks = listTasks;
/** Retrieves the specified task. */
function getTask(payload) {
    function __getTask({ host }) {
        const metadata = {
            entityFqdn: 'wix.data.autocms.v2.background_task',
            method: 'GET',
            methodFqn: 'wix.cloud.autocms.tasks.v1.TaskService.GetTask',
            packageName: PACKAGE_NAME,
            url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
                protoPath: '/v1/{taskId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'task.startedAt' },
                        { path: 'task.finishedAt' },
                        {
                            path: 'task.updatePublishStatusOptions.scheduleDraftStatusOptions.date',
                        },
                        {
                            path: 'task.updatePublishStatusOptions.schedulePublishedStatusOptions.date',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __getTask;
}
exports.getTask = getTask;
/** Cancels an existing task. */
function cancelTask(payload) {
    function __cancelTask({ host }) {
        const metadata = {
            entityFqdn: 'wix.data.autocms.v2.background_task',
            method: 'DELETE',
            methodFqn: 'wix.cloud.autocms.tasks.v1.TaskService.CancelTask',
            packageName: PACKAGE_NAME,
            url: resolveWixCloudAutocmsTasksV1TaskServiceUrl({
                protoPath: '/v1/{taskId}',
                data: payload,
                host,
            }),
            params: (0, rest_modules_1.toURLSearchParams)(payload),
            transformResponse: (payload) => (0, transform_paths_1.transformPaths)(payload, [
                {
                    transformFn: timestamp_2.transformRESTTimestampToSDKTimestamp,
                    paths: [
                        { path: 'task.startedAt' },
                        { path: 'task.finishedAt' },
                        {
                            path: 'task.updatePublishStatusOptions.scheduleDraftStatusOptions.date',
                        },
                        {
                            path: 'task.updatePublishStatusOptions.schedulePublishedStatusOptions.date',
                        },
                    ],
                },
            ]),
        };
        return metadata;
    }
    return __cancelTask;
}
exports.cancelTask = cancelTask;
