// @wix/auto_sdk_data_tasks/build/cjs/src/data-autocms-v2-background-task-tasks.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Type = exports.Status = exports.Operation = exports.Environment = exports.cancelTask = exports.getTask = exports.listTasks = exports.createTask = exports.__metadata = void 0;
const data_autocms_v2_background_task_tasks_universal_js_1 = require("./data-autocms-v2-background-task-tasks.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function createTask(httpClient) {
    return (task) => (0, data_autocms_v2_background_task_tasks_universal_js_1.createTask)(task, 
    // @ts-ignore
    { httpClient });
}
exports.createTask = createTask;
function listTasks(httpClient) {
    return (options) => (0, data_autocms_v2_background_task_tasks_universal_js_1.listTasks)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listTasks = listTasks;
function getTask(httpClient) {
    return (taskId) => (0, data_autocms_v2_background_task_tasks_universal_js_1.getTask)(taskId, 
    // @ts-ignore
    { httpClient });
}
exports.getTask = getTask;
function cancelTask(httpClient) {
    return (taskId) => (0, data_autocms_v2_background_task_tasks_universal_js_1.cancelTask)(taskId, 
    // @ts-ignore
    { httpClient });
}
exports.cancelTask = cancelTask;
var data_autocms_v2_background_task_tasks_universal_js_2 = require("./data-autocms-v2-background-task-tasks.universal.js");
Object.defineProperty(exports, "Environment", { enumerable: true, get: function () { return data_autocms_v2_background_task_tasks_universal_js_2.Environment; } });
Object.defineProperty(exports, "Operation", { enumerable: true, get: function () { return data_autocms_v2_background_task_tasks_universal_js_2.Operation; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return data_autocms_v2_background_task_tasks_universal_js_2.Status; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return data_autocms_v2_background_task_tasks_universal_js_2.Type; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return data_autocms_v2_background_task_tasks_universal_js_2.WebhookIdentityType; } });
