// @wix/auto_sdk_data_movement-job/build/cjs/src/hub-v1-job-movement-job.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WritePolicy = exports.WixDataEnvironment = exports.WebhookIdentityType = exports.Units = exports.SortOrder = exports.JobStatus = exports.FileFormat = exports.FieldType = exports.onJobUpdated = exports.onJobDeleted = exports.onJobCreated = exports.listMovementLogs = exports.terminateJob = exports.listJobs = exports.getJob = exports.createFileUploadUrl = exports.submitJob = exports.__metadata = void 0;
const rename_all_nested_keys_1 = require("@wix/sdk-runtime/rename-all-nested-keys");
const timestamp_1 = require("@wix/sdk-runtime/transformations/timestamp");
const transform_paths_1 = require("@wix/sdk-runtime/transformations/transform-paths");
const sdk_types_1 = require("@wix/sdk-types");
const hub_v1_job_movement_job_universal_js_1 = require("./hub-v1-job-movement-job.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function submitJob(httpClient) {
    return (source, destination, options) => (0, hub_v1_job_movement_job_universal_js_1.submitJob)(source, destination, options, 
    // @ts-ignore
    { httpClient });
}
exports.submitJob = submitJob;
function createFileUploadUrl(httpClient) {
    return (options) => (0, hub_v1_job_movement_job_universal_js_1.createFileUploadUrl)(options, 
    // @ts-ignore
    { httpClient });
}
exports.createFileUploadUrl = createFileUploadUrl;
function getJob(httpClient) {
    return (jobId) => (0, hub_v1_job_movement_job_universal_js_1.getJob)(jobId, 
    // @ts-ignore
    { httpClient });
}
exports.getJob = getJob;
function listJobs(httpClient) {
    return (options) => (0, hub_v1_job_movement_job_universal_js_1.listJobs)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listJobs = listJobs;
function terminateJob(httpClient) {
    return (jobId) => (0, hub_v1_job_movement_job_universal_js_1.terminateJob)(jobId, 
    // @ts-ignore
    { httpClient });
}
exports.terminateJob = terminateJob;
function listMovementLogs(httpClient) {
    return (jobId, options) => (0, hub_v1_job_movement_job_universal_js_1.listMovementLogs)(jobId, options, 
    // @ts-ignore
    { httpClient });
}
exports.listMovementLogs = listMovementLogs;
exports.onJobCreated = (0, sdk_types_1.EventDefinition)('wix.hub.v1.job_created', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startedAt' },
            { path: 'entity.finishedAt' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onJobDeleted = (0, sdk_types_1.EventDefinition)('wix.hub.v1.job_deleted', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.startedAt' },
            { path: 'undefined.finishedAt' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
exports.onJobUpdated = (0, sdk_types_1.EventDefinition)('wix.hub.v1.job_updated', true, (event) => (0, rename_all_nested_keys_1.renameKeysFromRESTResponseToSDKResponse)((0, transform_paths_1.transformPaths)(event, [
    {
        transformFn: timestamp_1.transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startedAt' },
            { path: 'entity.finishedAt' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
var hub_v1_job_movement_job_universal_js_2 = require("./hub-v1-job-movement-job.universal.js");
Object.defineProperty(exports, "FieldType", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.FieldType; } });
Object.defineProperty(exports, "FileFormat", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.FileFormat; } });
Object.defineProperty(exports, "JobStatus", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.JobStatus; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Units", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.Units; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.WebhookIdentityType; } });
Object.defineProperty(exports, "WixDataEnvironment", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.WixDataEnvironment; } });
Object.defineProperty(exports, "WritePolicy", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_2.WritePolicy; } });
