// @wix/auto_sdk_data_movement-job/build/cjs/src/hub-v1-job-movement-job.context.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortOrder = exports.WebhookIdentityType = exports.Units = exports.FieldType = exports.WritePolicy = exports.FileFormat = exports.WixDataEnvironment = exports.JobStatus = exports.onJobUpdated = exports.onJobDeleted = exports.onJobCreated = exports.listMovementLogs = exports.terminateJob = exports.listJobs = exports.getJob = exports.createFileUploadUrl = exports.submitJob = void 0;
const hub_v1_job_movement_job_public_js_1 = require("./hub-v1-job-movement-job.public.js");
const rest_modules_1 = require("@wix/sdk-runtime/rest-modules");
const event_definition_modules_1 = require("@wix/sdk-runtime/event-definition-modules");
const hub_v1_job_movement_job_public_js_2 = require("./hub-v1-job-movement-job.public.js");
const hub_v1_job_movement_job_public_js_3 = require("./hub-v1-job-movement-job.public.js");
const hub_v1_job_movement_job_public_js_4 = require("./hub-v1-job-movement-job.public.js");
exports.submitJob = (0, rest_modules_1.createRESTModule)(hub_v1_job_movement_job_public_js_1.submitJob);
exports.createFileUploadUrl = (0, rest_modules_1.createRESTModule)(hub_v1_job_movement_job_public_js_1.createFileUploadUrl);
exports.getJob = (0, rest_modules_1.createRESTModule)(hub_v1_job_movement_job_public_js_1.getJob);
exports.listJobs = (0, rest_modules_1.createRESTModule)(hub_v1_job_movement_job_public_js_1.listJobs);
exports.terminateJob = (0, rest_modules_1.createRESTModule)(hub_v1_job_movement_job_public_js_1.terminateJob);
exports.listMovementLogs = (0, rest_modules_1.createRESTModule)(hub_v1_job_movement_job_public_js_1.listMovementLogs);
/**
 * Triggered when a job is submitted.
 */
exports.onJobCreated = (0, event_definition_modules_1.createEventModule)(hub_v1_job_movement_job_public_js_2.onJobCreated);
/**
 * Triggered when a job is terminated.
 */
exports.onJobDeleted = (0, event_definition_modules_1.createEventModule)(hub_v1_job_movement_job_public_js_3.onJobDeleted);
/**
 * Triggered when a job's `status` updates.
 */
exports.onJobUpdated = (0, event_definition_modules_1.createEventModule)(hub_v1_job_movement_job_public_js_4.onJobUpdated);
var hub_v1_job_movement_job_universal_js_1 = require("./hub-v1-job-movement-job.universal.js");
Object.defineProperty(exports, "JobStatus", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.JobStatus; } });
Object.defineProperty(exports, "WixDataEnvironment", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.WixDataEnvironment; } });
Object.defineProperty(exports, "FileFormat", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.FileFormat; } });
Object.defineProperty(exports, "WritePolicy", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.WritePolicy; } });
Object.defineProperty(exports, "FieldType", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.FieldType; } });
Object.defineProperty(exports, "Units", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.Units; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.WebhookIdentityType; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return hub_v1_job_movement_job_universal_js_1.SortOrder; } });
