// @wix/auto_sdk_data_movement-job/build/cjs/src/hub-v1-job-movement-job.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SortOrder = exports.WebhookIdentityType = exports.Units = exports.FieldType = exports.WritePolicy = exports.FileFormat = exports.WixDataEnvironment = exports.JobStatus = void 0;
var JobStatus;
(function (JobStatus) {
    /** Job was submitted but has not yet started. */
    JobStatus["UNINITIALIZED"] = "UNINITIALIZED";
    /** Job is initializing. */
    JobStatus["INITIALIZING"] = "INITIALIZING";
    /** Job is in progress. */
    JobStatus["IN_PROGRESS"] = "IN_PROGRESS";
    /** Job has successfully completed. */
    JobStatus["COMPLETED"] = "COMPLETED";
    /** Job has failed. */
    JobStatus["FAILED"] = "FAILED";
    /** Job has completed but was partially successful due to a destination limitation. See `PartiallySuccessfulOptions` for additional details. */
    JobStatus["PARTIALLY_SUCCESSFUL"] = "PARTIALLY_SUCCESSFUL";
})(JobStatus || (exports.JobStatus = JobStatus = {}));
var WixDataEnvironment;
(function (WixDataEnvironment) {
    /** Collection's [live environment](https://support.wix.com/en/article/cms-about-sandbox-and-live-collections-and-syncing#sandbox-collections). */
    WixDataEnvironment["LIVE"] = "LIVE";
    /** Collection's [sandbox environment](https://support.wix.com/en/article/cms-about-sandbox-and-live-collections-and-syncing#live-collections). */
    WixDataEnvironment["SANDBOX"] = "SANDBOX";
})(WixDataEnvironment || (exports.WixDataEnvironment = WixDataEnvironment = {}));
var FileFormat;
(function (FileFormat) {
    /** Unknown format. */
    FileFormat["UNKNOWN"] = "UNKNOWN";
    /** CSV format. */
    FileFormat["CSV"] = "CSV";
    /** JSONL format. */
    FileFormat["JSONL"] = "JSONL";
})(FileFormat || (exports.FileFormat = FileFormat = {}));
var WritePolicy;
(function (WritePolicy) {
    /** When an item with the same ID exists in the collection, overwrite it. */
    WritePolicy["OVERWRITE"] = "OVERWRITE";
    /** When an item with the same ID exists in the collection, skip it. */
    WritePolicy["SKIP_EXISTING"] = "SKIP_EXISTING";
    /** Clear the destination collection before moving data. When the job completes, the collection contains only the moved data. */
    WritePolicy["TRUNCATE_BEFORE"] = "TRUNCATE_BEFORE";
})(WritePolicy || (exports.WritePolicy = WritePolicy = {}));
var FieldType;
(function (FieldType) {
    /** Any data type. */
    FieldType["ANY"] = "ANY";
    /** String. */
    FieldType["STRING"] = "STRING";
    /** Number. */
    FieldType["NUMBER"] = "NUMBER";
    /** Boolean. */
    FieldType["BOOLEAN"] = "BOOLEAN";
    /** Date string in ISO 8601 date format: `YYYY-MM-DD`. */
    FieldType["DATE"] = "DATE";
    /** [Date object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date) (SDK) or an object in the following format: `"someFieldKey": { "$date": "YYYY-MM-DDTHH:mm:ss.sssZ"}`. */
    FieldType["DATETIME"] = "DATETIME";
    /** Binary file. */
    FieldType["BINARY"] = "BINARY";
    /** object. */
    FieldType["OBJECT"] = "OBJECT";
    /** Array. */
    FieldType["ARRAY"] = "ARRAY";
    /** Web URL. */
    FieldType["URL"] = "URL";
    FieldType["WIX_MEDIA_URL"] = "WIX_MEDIA_URL";
    FieldType["WIX_MEDIA_IMAGE_URL"] = "WIX_MEDIA_IMAGE_URL";
    FieldType["WIX_MEDIA_VIDEO_URL"] = "WIX_MEDIA_VIDEO_URL";
    FieldType["WIX_MEDIA_DOCUMENT_URL"] = "WIX_MEDIA_DOCUMENT_URL";
    FieldType["IMAGE_URL"] = "IMAGE_URL";
    FieldType["VIDEO_URL"] = "VIDEO_URL";
})(FieldType || (exports.FieldType = FieldType = {}));
var Units;
(function (Units) {
    /** Default value. This value is unused. */
    Units["UNKNOWN_UNITS"] = "UNKNOWN_UNITS";
    /** Unit of bytes. */
    Units["BYTES"] = "BYTES";
    /** Unit of items. */
    Units["ITEMS"] = "ITEMS";
})(Units || (exports.Units = Units = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
var SortOrder;
(function (SortOrder) {
    SortOrder["ASC"] = "ASC";
    SortOrder["DESC"] = "DESC";
})(SortOrder || (exports.SortOrder = SortOrder = {}));
