// @wix/auto_sdk_data_collections/build/internal/cjs/src/data-v2-data-collection-collections.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteDataCollectionField = exports.updateDataCollectionField = exports.createDataCollectionField = exports.restoreDataCollection = exports.deleteDataCollection = exports.updateDataCollection = exports.listDataCollections = exports.getDataCollection = exports.createDataCollection = void 0;
const ambassadorWixDataV2DataCollection = __importStar(require("./data-v2-data-collection-collections.http.js"));
function createDataCollection() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.createDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDataCollection = createDataCollection;
function getDataCollection() {
    const payload = { dataCollectionId: ':dataCollectionId' };
    const getRequestOptions = ambassadorWixDataV2DataCollection.getDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections/{dataCollectionId}',
        pathParams: { dataCollectionId: 'dataCollectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getDataCollection = getDataCollection;
function listDataCollections() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.listDataCollections(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listDataCollections = listDataCollections;
function updateDataCollection() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.updateDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'PUT',
        path: '/v2/collections',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDataCollection = updateDataCollection;
function deleteDataCollection() {
    const payload = { dataCollectionId: ':dataCollectionId' };
    const getRequestOptions = ambassadorWixDataV2DataCollection.deleteDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/collections/{dataCollectionId}',
        pathParams: { dataCollectionId: 'dataCollectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteDataCollection = deleteDataCollection;
function restoreDataCollection() {
    const payload = { dataCollectionId: ':dataCollectionId' };
    const getRequestOptions = ambassadorWixDataV2DataCollection.restoreDataCollection(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/restore/{dataCollectionId}',
        pathParams: { dataCollectionId: 'dataCollectionId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.restoreDataCollection = restoreDataCollection;
function createDataCollectionField() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.createDataCollectionField(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/create-field',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createDataCollectionField = createDataCollectionField;
function updateDataCollectionField() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.updateDataCollectionField(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/update-field',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateDataCollectionField = updateDataCollectionField;
function deleteDataCollectionField() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2DataCollection.deleteDataCollectionField(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/collections/delete-field',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteDataCollectionField = deleteDataCollectionField;
