// @wix/auto_sdk_data_collections/build/internal/cjs/src/data-v2-data-collection-collections.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Type = exports.Status = exports.SortOrder = exports.Segment = exports.Role = exports.QueryOperator = exports.PluginType = exports.PagingMode = exports.Order = exports.IndexStatus = exports.IndexSource = exports.Format = exports.FieldPluginType = exports.Direction = exports.DataOperation = exports.CollectionType = exports.CollectionOperation = exports.AccessLevel = exports.deleteDataCollectionField = exports.updateDataCollectionField = exports.createDataCollectionField = exports.restoreDataCollection = exports.deleteDataCollection = exports.updateDataCollection = exports.listDataCollections = exports.getDataCollection = exports.createDataCollection = exports.__metadata = void 0;
const data_v2_data_collection_collections_universal_js_1 = require("./data-v2-data-collection-collections.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function createDataCollection(httpClient) {
    return (collection) => (0, data_v2_data_collection_collections_universal_js_1.createDataCollection)(collection, 
    // @ts-ignore
    { httpClient });
}
exports.createDataCollection = createDataCollection;
function getDataCollection(httpClient) {
    return (dataCollectionId, options) => (0, data_v2_data_collection_collections_universal_js_1.getDataCollection)(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.getDataCollection = getDataCollection;
function listDataCollections(httpClient) {
    return (options) => (0, data_v2_data_collection_collections_universal_js_1.listDataCollections)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listDataCollections = listDataCollections;
function updateDataCollection(httpClient) {
    return (collection) => (0, data_v2_data_collection_collections_universal_js_1.updateDataCollection)(collection, 
    // @ts-ignore
    { httpClient });
}
exports.updateDataCollection = updateDataCollection;
function deleteDataCollection(httpClient) {
    return (dataCollectionId) => (0, data_v2_data_collection_collections_universal_js_1.deleteDataCollection)(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteDataCollection = deleteDataCollection;
/** @internal */
function restoreDataCollection(httpClient) {
    return (dataCollectionId) => (0, data_v2_data_collection_collections_universal_js_1.restoreDataCollection)(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
exports.restoreDataCollection = restoreDataCollection;
/** @internal */
function createDataCollectionField(httpClient) {
    return (dataCollectionId, options) => (0, data_v2_data_collection_collections_universal_js_1.createDataCollectionField)(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.createDataCollectionField = createDataCollectionField;
/** @internal */
function updateDataCollectionField(httpClient) {
    return (dataCollectionId, options) => (0, data_v2_data_collection_collections_universal_js_1.updateDataCollectionField)(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.updateDataCollectionField = updateDataCollectionField;
/** @internal */
function deleteDataCollectionField(httpClient) {
    return (dataCollectionId, options) => (0, data_v2_data_collection_collections_universal_js_1.deleteDataCollectionField)(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
exports.deleteDataCollectionField = deleteDataCollectionField;
var data_v2_data_collection_collections_universal_js_2 = require("./data-v2-data-collection-collections.universal.js");
Object.defineProperty(exports, "AccessLevel", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.AccessLevel; } });
Object.defineProperty(exports, "CollectionOperation", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.CollectionOperation; } });
Object.defineProperty(exports, "CollectionType", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.CollectionType; } });
Object.defineProperty(exports, "DataOperation", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.DataOperation; } });
Object.defineProperty(exports, "Direction", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Direction; } });
Object.defineProperty(exports, "FieldPluginType", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.FieldPluginType; } });
Object.defineProperty(exports, "Format", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Format; } });
Object.defineProperty(exports, "IndexSource", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.IndexSource; } });
Object.defineProperty(exports, "IndexStatus", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.IndexStatus; } });
Object.defineProperty(exports, "Order", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Order; } });
Object.defineProperty(exports, "PagingMode", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.PagingMode; } });
Object.defineProperty(exports, "PluginType", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.PluginType; } });
Object.defineProperty(exports, "QueryOperator", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.QueryOperator; } });
Object.defineProperty(exports, "Role", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Role; } });
Object.defineProperty(exports, "Segment", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Segment; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.SortOrder; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Status; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.Type; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return data_v2_data_collection_collections_universal_js_2.WebhookIdentityType; } });
