// @wix/auto_sdk_data_tasks/build/internal/cjs/src/data-autocms-v2-background-task-tasks.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.cancelTask = exports.getTask = exports.listTasks = exports.createTask = void 0;
const ambassadorWixDataAutocmsV2BackgroundTask = __importStar(require("./data-autocms-v2-background-task-tasks.http.js"));
function createTask() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.createTask(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createTask = createTask;
function listTasks() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.listTasks(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listTasks = listTasks;
function getTask() {
    const payload = { taskId: ':taskId' };
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.getTask(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/{taskId}',
        pathParams: { taskId: 'taskId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getTask = getTask;
function cancelTask() {
    const payload = { taskId: ':taskId' };
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.cancelTask(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/{taskId}',
        pathParams: { taskId: 'taskId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.cancelTask = cancelTask;
