// @wix/auto_sdk_data_tasks/build/internal/cjs/src/data-autocms-v2-background-task-tasks.types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Operation = exports.Environment = exports.Status = exports.Type = void 0;
var Type;
(function (Type) {
    /** Unknown task type. */
    Type["UNKNOWN"] = "UNKNOWN";
    /** Deletes all items that match the specified filter. */
    Type["DELETE_BY_QUERY"] = "DELETE_BY_QUERY";
    /** Copies data from one field to another for all items in the collection. */
    Type["COPY_FIELD_DATA"] = "COPY_FIELD_DATA";
    /** Updates field data for items that match the specified filter. */
    Type["EDIT_FIELD_DATA"] = "EDIT_FIELD_DATA";
    /** In collections that have the Publish plugin, updates the publish status of items that match the specified filter. */
    Type["UPDATE_PUBLISH_STATUS"] = "UPDATE_PUBLISH_STATUS";
})(Type || (exports.Type = Type = {}));
var Status;
(function (Status) {
    /** Task was created but has not started. */
    Status["NEW"] = "NEW";
    /** Task is currently running. */
    Status["RUNNING"] = "RUNNING";
    /** Task has completed. Partial errors are listed in `failures`. */
    Status["COMPLETED"] = "COMPLETED";
    /**
     * Task has failed. Tasks can fail due to errors or because they were cancelled.
     *
     * Failures are listed in `failures`. Cancelled tasks have a failure `code` value of `CANCELLED`, and no associated failure `data`.
     */
    Status["FAILED"] = "FAILED";
})(Status || (exports.Status = Status = {}));
var Environment;
(function (Environment) {
    /** Place holder. Never returned by the service. */
    Environment["UNKNOWN_ENVIRONMENT"] = "UNKNOWN_ENVIRONMENT";
    /** Collection [sandbox environment](https://support.wix.com/en/article/cms-about-sandbox-and-live-collections-and-syncing#sandbox-collections). */
    Environment["SANDBOX"] = "SANDBOX";
    /** If the [collection's sandbox environment is enabled](https://support.wix.com/en/article/cms-accessing-and-syncing-your-sandbox-and-live-collections#enabling-sandbox-collections), run the task in the collection's sandbox environment. If not, run the task in the collection's live environment. */
    Environment["SANDBOX_PREFERRED"] = "SANDBOX_PREFERRED";
    /** Collection [live environment](https://support.wix.com/en/article/cms-about-sandbox-and-live-collections-and-syncing#live-collections). */
    Environment["LIVE"] = "LIVE";
})(Environment || (exports.Environment = Environment = {}));
var Operation;
(function (Operation) {
    Operation["UNDEFINED"] = "UNDEFINED";
    /** Update the publish status of all items that match the filter to `PUBLISHED`. */
    Operation["SET_PUBLISHED_STATUS"] = "SET_PUBLISHED_STATUS";
    /** Update the publish status of all items that match the filter to `DRAFT`. */
    Operation["SET_DRAFT_STATUS"] = "SET_DRAFT_STATUS";
    /** Schedule when to update the publish status of all items that match the filter to `PUBLISHED`. */
    Operation["SCHEDULE_PUBLISHED_STATUS"] = "SCHEDULE_PUBLISHED_STATUS";
    /** Schedule when to update the publish status of all items that match the filter to `DRAFT`. */
    Operation["SCHEDULE_DRAFT_STATUS"] = "SCHEDULE_DRAFT_STATUS";
    /** Cancel all scheduled status updates. */
    Operation["CANCEL_SCHEDULING"] = "CANCEL_SCHEDULING";
})(Operation || (exports.Operation = Operation = {}));
var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (exports.WebhookIdentityType = WebhookIdentityType = {}));
