// @wix/auto_sdk_data_collections/build/es/src/data-v2-data-collection-collections.public.js
import { createDataCollection as universalCreateDataCollection, createDataCollectionField as universalCreateDataCollectionField, deleteDataCollection as universalDeleteDataCollection, deleteDataCollectionField as universalDeleteDataCollectionField, getDataCollection as universalGetDataCollection, listDataCollections as universalListDataCollections, restoreDataCollection as universalRestoreDataCollection, updateDataCollection as universalUpdateDataCollection, updateDataCollectionField as universalUpdateDataCollectionField, } from './data-v2-data-collection-collections.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/data' };
export function createDataCollection(httpClient) {
    return (collection) => universalCreateDataCollection(collection, 
    // @ts-ignore
    { httpClient });
}
export function getDataCollection(httpClient) {
    return (dataCollectionId, options) => universalGetDataCollection(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
export function listDataCollections(httpClient) {
    return (options) => universalListDataCollections(options, 
    // @ts-ignore
    { httpClient });
}
export function updateDataCollection(httpClient) {
    return (collection) => universalUpdateDataCollection(collection, 
    // @ts-ignore
    { httpClient });
}
export function deleteDataCollection(httpClient) {
    return (dataCollectionId) => universalDeleteDataCollection(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
/** @internal */
export function restoreDataCollection(httpClient) {
    return (dataCollectionId) => universalRestoreDataCollection(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
/** @internal */
export function createDataCollectionField(httpClient) {
    return (dataCollectionId, options) => universalCreateDataCollectionField(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
/** @internal */
export function updateDataCollectionField(httpClient) {
    return (dataCollectionId, options) => universalUpdateDataCollectionField(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
/** @internal */
export function deleteDataCollectionField(httpClient) {
    return (dataCollectionId, options) => universalDeleteDataCollectionField(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
export { AccessLevel, CollectionOperation, CollectionType, DataOperation, Direction, FieldPluginType, Format, IndexSource, IndexStatus, Order, PagingMode, PluginType, QueryOperator, Role, Segment, SortOrder, Status, Type, WebhookIdentityType, } from './data-v2-data-collection-collections.universal.js';
