// @wix/auto_sdk_data_indexes/build/es/src/data-v2-index-indexes.public.js
import { createIndex as universalCreateIndex, dropIndex as universalDropIndex, listAvailableIndexes as universalListAvailableIndexes, listIndexes as universalListIndexes, } from './data-v2-index-indexes.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/data' };
export function createIndex(httpClient) {
    return (dataCollectionId, index) => universalCreateIndex(dataCollectionId, index, 
    // @ts-ignore
    { httpClient });
}
export function dropIndex(httpClient) {
    return (dataCollectionId, indexName) => universalDropIndex(dataCollectionId, indexName, 
    // @ts-ignore
    { httpClient });
}
export function listIndexes(httpClient) {
    return (dataCollectionId, options) => universalListIndexes(dataCollectionId, options, 
    // @ts-ignore
    { httpClient });
}
/** @internal */
export function listAvailableIndexes(httpClient) {
    return (dataCollectionId) => universalListAvailableIndexes(dataCollectionId, 
    // @ts-ignore
    { httpClient });
}
export { Environment, IndexSource, Order, Status, } from './data-v2-index-indexes.universal.js';
