// @wix/auto_sdk_data_permissions/build/cjs/src/data-v1-data-permissions-permissions.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getMyPermissions = exports.removeSpecialPermissions = exports.updateSpecialPermissions = exports.addSpecialPermissions = exports.updatePermissions = exports.getPermissions = void 0;
const ambassadorWixDataV1DataPermissions = __importStar(require("./data-v1-data-permissions-permissions.http.js"));
function getPermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.getPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/permissions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getPermissions = getPermissions;
function updatePermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.updatePermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/permissions',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updatePermissions = updatePermissions;
function addSpecialPermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.addSpecialPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/permissions/special',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.addSpecialPermissions = addSpecialPermissions;
function updateSpecialPermissions() {
    const payload = {
        specialPermissions: { id: ':specialPermissionsId' },
    };
    const getRequestOptions = ambassadorWixDataV1DataPermissions.updateSpecialPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/permissions/special/{specialPermissions.id}',
        pathParams: { specialPermissionsId: 'specialPermissionsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.updateSpecialPermissions = updateSpecialPermissions;
function removeSpecialPermissions() {
    const payload = { specialPermissionsId: ':specialPermissionsId' };
    const getRequestOptions = ambassadorWixDataV1DataPermissions.removeSpecialPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/permissions/special/{specialPermissionsId}',
        pathParams: { specialPermissionsId: 'specialPermissionsId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.removeSpecialPermissions = removeSpecialPermissions;
function getMyPermissions() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV1DataPermissions.getMyPermissions(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/permissions/current',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.getMyPermissions = getMyPermissions;
