// @wix/auto_sdk_data_backups/build/es/src/data-v2-backup-backups.types.js
export var Status;
(function (Status) {
    /** Backup creation is in progress. */
    Status["PENDING"] = "PENDING";
    /** Backup has been created successfully and can be used for data restoration. */
    Status["READY"] = "READY";
    /** Backup creation has failed. */
    Status["FAILED"] = "FAILED";
    /** Backup has been deleted. */
    Status["DELETED"] = "DELETED";
    /** Backup has been canceled. */
    Status["CANCELED"] = "CANCELED";
})(Status || (Status = {}));
export var Type;
(function (Type) {
    /** Backup taken on demand. */
    Type["ON_DEMAND"] = "ON_DEMAND";
    /** Backup taken automatically by the system on a regular schedule. */
    Type["AUTO"] = "AUTO";
})(Type || (Type = {}));
export var RestorationStatus;
(function (RestorationStatus) {
    /** Restoration from a backup is in progress. */
    RestorationStatus["PENDING"] = "PENDING";
    /** Restoration from a backup has been successful. */
    RestorationStatus["COMPLETED"] = "COMPLETED";
    /** Restoration from a backup has failed. */
    RestorationStatus["FAILED"] = "FAILED";
})(RestorationStatus || (RestorationStatus = {}));
export var WebhookIdentityType;
(function (WebhookIdentityType) {
    WebhookIdentityType["UNKNOWN"] = "UNKNOWN";
    WebhookIdentityType["ANONYMOUS_VISITOR"] = "ANONYMOUS_VISITOR";
    WebhookIdentityType["MEMBER"] = "MEMBER";
    WebhookIdentityType["WIX_USER"] = "WIX_USER";
    WebhookIdentityType["APP"] = "APP";
})(WebhookIdentityType || (WebhookIdentityType = {}));
