// @wix/auto_sdk_data_movement-job/build/es/src/hub-v1-job-movement-job.meta.js
import * as ambassadorWixHubV1Job from './hub-v1-job-movement-job.http.js';
export function submitJob() {
    const payload = {};
    const getRequestOptions = ambassadorWixHubV1Job.submitJob(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/jobs',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function createFileUploadUrl() {
    const payload = {};
    const getRequestOptions = ambassadorWixHubV1Job.createFileUploadUrl(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/temporary-files/upload-url',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getJob() {
    const payload = { jobId: ':jobId' };
    const getRequestOptions = ambassadorWixHubV1Job.getJob(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/jobs/{jobId}',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listJobs() {
    const payload = {};
    const getRequestOptions = ambassadorWixHubV1Job.listJobs(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/jobs',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function terminateJob() {
    const payload = { jobId: ':jobId' };
    const getRequestOptions = ambassadorWixHubV1Job.terminateJob(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1/jobs/{jobId}/terminate',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listMovementLogs() {
    const payload = { jobId: ':jobId' };
    const getRequestOptions = ambassadorWixHubV1Job.listMovementLogs(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/jobs/{jobId}/logs',
        pathParams: { jobId: 'jobId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
