// @wix/auto_sdk_data_movement-job/build/es/src/hub-v1-job-movement-job.http.js
import { toURLSearchParams } from '@wix/sdk-runtime/rest-modules';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { resolveUrl } from '@wix/sdk-runtime/rest-modules';
function resolveWixHubV1DataMovementServiceUrl(opts) {
    const domainToMappings = {
        'manage._base_domain_': [
            {
                srcPath: '/_api/wix-data-hub-fs2',
                destPath: '',
            },
        ],
        'editor._base_domain_': [
            {
                srcPath: '/_api/wix-data-hub-fs2',
                destPath: '',
            },
        ],
        'blocks._base_domain_': [
            {
                srcPath: '/_api/wix-data-hub-fs2',
                destPath: '',
            },
        ],
        'create.editorx': [
            {
                srcPath: '/_api/wix-data-hub-fs2',
                destPath: '',
            },
        ],
        'www.wixapis.com': [
            {
                srcPath: '/data-mover',
                destPath: '',
            },
        ],
        '*.dev.wix-code.com': [
            {
                srcPath: '/_api/wix-data-hub-fs2',
                destPath: '',
            },
        ],
        _: [
            {
                srcPath: '/_api/wix-data-hub-fs2',
                destPath: '',
            },
        ],
    };
    return resolveUrl(Object.assign(opts, { domainToMappings }));
}
const PACKAGE_NAME = '@wix/auto_sdk_data_movement-job';
/**
 * Submit a data movement job.
 *
 * > **Note**: By default, the job is submitted for the current site.
 */
export function submitJob(payload) {
    function __submitJob({ host }) {
        const metadata = {
            entityFqdn: 'wix.hub.v1.job',
            method: 'POST',
            methodFqn: 'wix.hub.v1.DataMovementService.SubmitJob',
            packageName: PACKAGE_NAME,
            url: resolveWixHubV1DataMovementServiceUrl({
                protoPath: '/v1/jobs',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'job.startedAt' }, { path: 'job.finishedAt' }],
                },
            ]),
        };
        return metadata;
    }
    return __submitJob;
}
/**
 * Creates a URL to upload a data source file. The URL is valid for 15 minutes.
 *
 * Call this method when submitting a job to import data from a file to a site.
 */
export function createFileUploadUrl(payload) {
    function __createFileUploadUrl({ host }) {
        const metadata = {
            entityFqdn: 'wix.hub.v1.job',
            method: 'POST',
            methodFqn: 'wix.hub.v1.DataMovementService.CreateFileUploadUrl',
            packageName: PACKAGE_NAME,
            url: resolveWixHubV1DataMovementServiceUrl({
                protoPath: '/v1/temporary-files/upload-url',
                data: payload,
                host,
            }),
            data: payload,
        };
        return metadata;
    }
    return __createFileUploadUrl;
}
/**
 * Retrieves the specified data movement job.
 *
 * > **Note**: You can only retrieve jobs submitted for the current site.
 */
export function getJob(payload) {
    function __getJob({ host }) {
        const metadata = {
            entityFqdn: 'wix.hub.v1.job',
            method: 'GET',
            methodFqn: 'wix.hub.v1.DataMovementService.GetJob',
            packageName: PACKAGE_NAME,
            url: resolveWixHubV1DataMovementServiceUrl({
                protoPath: '/v1/jobs/{jobId}',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'job.startedAt' }, { path: 'job.finishedAt' }],
                },
            ]),
        };
        return metadata;
    }
    return __getJob;
}
/** Lists data movement jobs for the current site. */
export function listJobs(payload) {
    function __listJobs({ host }) {
        const metadata = {
            entityFqdn: 'wix.hub.v1.job',
            method: 'POST',
            methodFqn: 'wix.hub.v1.DataMovementService.ListJobs',
            packageName: PACKAGE_NAME,
            url: resolveWixHubV1DataMovementServiceUrl({
                protoPath: '/v1/jobs/list',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'jobs.startedAt' }, { path: 'jobs.finishedAt' }],
                },
            ]),
            fallback: [
                {
                    method: 'POST',
                    url: resolveWixHubV1DataMovementServiceUrl({
                        protoPath: '/v1/jobs/list',
                        data: payload,
                        host,
                    }),
                    data: payload,
                },
            ],
        };
        return metadata;
    }
    return __listJobs;
}
/** Terminates the specified job. */
export function terminateJob(payload) {
    function __terminateJob({ host }) {
        const metadata = {
            entityFqdn: 'wix.hub.v1.job',
            method: 'POST',
            methodFqn: 'wix.hub.v1.DataMovementService.TerminateJob',
            packageName: PACKAGE_NAME,
            url: resolveWixHubV1DataMovementServiceUrl({
                protoPath: '/v1/jobs/{jobId}/terminate',
                data: payload,
                host,
            }),
            data: payload,
            transformResponse: (payload) => transformPaths(payload, [
                {
                    transformFn: transformRESTTimestampToSDKTimestamp,
                    paths: [{ path: 'job.startedAt' }, { path: 'job.finishedAt' }],
                },
            ]),
        };
        return metadata;
    }
    return __terminateJob;
}
/**
 * Lists logs for the specified job.
 *
 * Logs are generated by the destination for each item that failed to move. Some destinations also generate logs for items that moved successfully.
 */
export function listMovementLogs(payload) {
    function __listMovementLogs({ host }) {
        const metadata = {
            entityFqdn: 'wix.hub.v1.job',
            method: 'GET',
            methodFqn: 'wix.hub.v1.DataMovementService.ListMovementLogs',
            packageName: PACKAGE_NAME,
            url: resolveWixHubV1DataMovementServiceUrl({
                protoPath: '/v1/jobs/{jobId}/logs',
                data: payload,
                host,
            }),
            params: toURLSearchParams(payload),
        };
        return metadata;
    }
    return __listMovementLogs;
}
