// @wix/auto_sdk_data_movement-job/build/es/src/hub-v1-job-movement-job.public.js
import { renameKeysFromRESTResponseToSDKResponse } from '@wix/sdk-runtime/rename-all-nested-keys';
import { transformRESTTimestampToSDKTimestamp } from '@wix/sdk-runtime/transformations/timestamp';
import { transformPaths } from '@wix/sdk-runtime/transformations/transform-paths';
import { EventDefinition } from '@wix/sdk-types';
import { createFileUploadUrl as universalCreateFileUploadUrl, getJob as universalGetJob, listJobs as universalListJobs, listMovementLogs as universalListMovementLogs, submitJob as universalSubmitJob, terminateJob as universalTerminateJob, } from './hub-v1-job-movement-job.universal.js';
export const __metadata = { PACKAGE_NAME: '@wix/data' };
export function submitJob(httpClient) {
    return (source, destination, options) => universalSubmitJob(source, destination, options, 
    // @ts-ignore
    { httpClient });
}
export function createFileUploadUrl(httpClient) {
    return (options) => universalCreateFileUploadUrl(options, 
    // @ts-ignore
    { httpClient });
}
export function getJob(httpClient) {
    return (jobId) => universalGetJob(jobId, 
    // @ts-ignore
    { httpClient });
}
export function listJobs(httpClient) {
    return (options) => universalListJobs(options, 
    // @ts-ignore
    { httpClient });
}
export function terminateJob(httpClient) {
    return (jobId) => universalTerminateJob(jobId, 
    // @ts-ignore
    { httpClient });
}
export function listMovementLogs(httpClient) {
    return (jobId, options) => universalListMovementLogs(jobId, options, 
    // @ts-ignore
    { httpClient });
}
export const onJobCreated = EventDefinition('wix.hub.v1.job_created', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startedAt' },
            { path: 'entity.finishedAt' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onJobDeleted = EventDefinition('wix.hub.v1.job_deleted', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'undefined.startedAt' },
            { path: 'undefined.finishedAt' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export const onJobUpdated = EventDefinition('wix.hub.v1.job_updated', true, (event) => renameKeysFromRESTResponseToSDKResponse(transformPaths(event, [
    {
        transformFn: transformRESTTimestampToSDKTimestamp,
        paths: [
            { path: 'entity.startedAt' },
            { path: 'entity.finishedAt' },
            { path: 'metadata.eventTime' },
        ],
    },
])))();
export { FieldType, FileFormat, JobStatus, SortOrder, Units, WebhookIdentityType, WixDataEnvironment, WritePolicy, } from './hub-v1-job-movement-job.universal.js';
