// @wix/auto_sdk_data_backups/build/internal/cjs/src/data-v2-backup-backups.public.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WebhookIdentityType = exports.Type = exports.Status = exports.RestorationStatus = exports.deleteBackup = exports.listRestorations = exports.restorePartialBackup = exports.restoreBackup = exports.listBackups = exports.createBackup = exports.__metadata = void 0;
const data_v2_backup_backups_universal_js_1 = require("./data-v2-backup-backups.universal.js");
exports.__metadata = { PACKAGE_NAME: '@wix/data' };
function createBackup(httpClient) {
    return () => (0, data_v2_backup_backups_universal_js_1.createBackup)(
    // @ts-ignore
    { httpClient });
}
exports.createBackup = createBackup;
function listBackups(httpClient) {
    return (options) => (0, data_v2_backup_backups_universal_js_1.listBackups)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listBackups = listBackups;
function restoreBackup(httpClient) {
    return (backupId) => (0, data_v2_backup_backups_universal_js_1.restoreBackup)(backupId, 
    // @ts-ignore
    { httpClient });
}
exports.restoreBackup = restoreBackup;
function restorePartialBackup(httpClient) {
    return (backupId, restorationCollections) => (0, data_v2_backup_backups_universal_js_1.restorePartialBackup)(backupId, restorationCollections, 
    // @ts-ignore
    { httpClient });
}
exports.restorePartialBackup = restorePartialBackup;
function listRestorations(httpClient) {
    return (options) => (0, data_v2_backup_backups_universal_js_1.listRestorations)(options, 
    // @ts-ignore
    { httpClient });
}
exports.listRestorations = listRestorations;
function deleteBackup(httpClient) {
    return (backupId) => (0, data_v2_backup_backups_universal_js_1.deleteBackup)(backupId, 
    // @ts-ignore
    { httpClient });
}
exports.deleteBackup = deleteBackup;
var data_v2_backup_backups_universal_js_2 = require("./data-v2-backup-backups.universal.js");
Object.defineProperty(exports, "RestorationStatus", { enumerable: true, get: function () { return data_v2_backup_backups_universal_js_2.RestorationStatus; } });
Object.defineProperty(exports, "Status", { enumerable: true, get: function () { return data_v2_backup_backups_universal_js_2.Status; } });
Object.defineProperty(exports, "Type", { enumerable: true, get: function () { return data_v2_backup_backups_universal_js_2.Type; } });
Object.defineProperty(exports, "WebhookIdentityType", { enumerable: true, get: function () { return data_v2_backup_backups_universal_js_2.WebhookIdentityType; } });
