// @wix/auto_sdk_data_tasks/build/es/src/data-autocms-v2-background-task-tasks.meta.js
import * as ambassadorWixDataAutocmsV2BackgroundTask from './data-autocms-v2-background-task-tasks.http.js';
export function createTask() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.createTask(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v1',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function listTasks() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.listTasks(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function getTask() {
    const payload = { taskId: ':taskId' };
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.getTask(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v1/{taskId}',
        pathParams: { taskId: 'taskId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
export function cancelTask() {
    const payload = { taskId: ':taskId' };
    const getRequestOptions = ambassadorWixDataAutocmsV2BackgroundTask.cancelTask(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v1/{taskId}',
        pathParams: { taskId: 'taskId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
