// @wix/auto_sdk_data_backups/build/cjs/src/data-v2-backup-backups.meta.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deleteBackup = exports.listRestorations = exports.restorePartialBackup = exports.restoreBackup = exports.listBackups = exports.createBackup = void 0;
const ambassadorWixDataV2Backup = __importStar(require("./data-v2-backup-backups.http.js"));
function createBackup() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Backup.createBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/backups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.createBackup = createBackup;
function listBackups() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Backup.listBackups(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/backups',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listBackups = listBackups;
function restoreBackup() {
    const payload = { backupId: ':backupId' };
    const getRequestOptions = ambassadorWixDataV2Backup.restoreBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/backups/{backupId}/restore',
        pathParams: { backupId: 'backupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.restoreBackup = restoreBackup;
function restorePartialBackup() {
    const payload = { backupId: ':backupId' };
    const getRequestOptions = ambassadorWixDataV2Backup.restorePartialBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'POST',
        path: '/v2/backups/{backupId}/partial-restore',
        pathParams: { backupId: 'backupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.restorePartialBackup = restorePartialBackup;
function listRestorations() {
    const payload = {};
    const getRequestOptions = ambassadorWixDataV2Backup.listRestorations(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'GET',
        path: '/v2/backups/restorations',
        pathParams: {},
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.listRestorations = listRestorations;
function deleteBackup() {
    const payload = { backupId: ':backupId' };
    const getRequestOptions = ambassadorWixDataV2Backup.deleteBackup(payload);
    const getUrl = (context) => {
        const { url } = getRequestOptions(context);
        return url;
    };
    return {
        getUrl,
        httpMethod: 'DELETE',
        path: '/v2/backups/{backupId}',
        pathParams: { backupId: 'backupId' },
        __requestType: null,
        __originalRequestType: null,
        __responseType: null,
        __originalResponseType: null,
    };
}
exports.deleteBackup = deleteBackup;
